### PR TITLE
Add type hints

### PR DESCRIPTION
**Pull request by anapaulagomes**

### CODE REVIEW

**Comment by anapaulagomes** [comment]
No rush for this one, @dmiele83. It can be reviewed on the guild day. :)

**Comment by dmiele83** [comment]
@anapaulagomes ok :)

**Comment by anapaulagomes** [comment]
Friendly reminder here @dmiele83 :)

File: Pipfile [hunk 1]
```diff
@@ -20,3 +21,6 @@ pydocstyle = "*"
 isort = "*"
 pylint = "*"
 pytest = "*"
+pytest-mypy = "*"
+mypy = "*"
+typing-extensions = "*"
```
Comment (dmiele83) [inline Pipfile]: Do we need this extension? it seems that we use Python3.7 for this project and the documentation says that maybe we can skip it: https://pypi.org/project/typing-extensions/

Comment (anapaulagomes) [reply Pipfile]: Yes, we do. It's really weird but tests are failing without it (due to `mypy` dependencies, I believe). :/

Comment (dmiele83) [reply Pipfile]: yes, you are right :) https://github.com/python/mypy/blob/master/setup.py

Screen Shot 2019-11-08 at 12 05 10 PM

**Pull request merged**